---
핵심

호출된 매서드가 실패하더라도, 해당 객체는 메서드 호출 전 상태를 유지해야 한다.


---
오해

실패 원자성은 100%는 아닌게,
ex. 두 스레드가 동기화 없이 같은 객체를 동시에 수정한다면, 그 객체의 일관성이 깨질 수 있다.
따라서 ConcurrentModificationException을 잡아냈다고 해서 그 객체가 여전히 쓸 수 있는 상태라고 가정하면 안된다.

한편, Error은 복구할 수 없으므로, AssertError에 대해서는 실패 원자적으로 만들려는 시도조차 할 필요가 없다.
